examples/compose/plot_column_transformer_mixed_types.py

```diff
@@ -78,6 +82,46 @@
 clf.fit(X_train, y_train)
 print("model score: %.3f" % clf.score(X_test, y_test))
 
+###############################################################################
+# Use ``ColumnTransformer`` by selecting column by data types
+###############################################################################
+# When dealing with cleaned dataset, the preprocessing can be automatic by
+# using the data types of the column to decide whether to treat a column as a
+# numerical or categorical features.
+# :func:`sklearn.compose.make_column_selector` gives this possibility.
+# First, let's only select a subset of columns to ease the simplify our
+# example.
+
+subset_feature = ['embarked', 'sex', 'pclass', 'age', 'fare']
+X = X[subset_feature]
+
+###############################################################################
+# Then, we introspect the information regarding each column data type.
+
+print(X.info())
+
+###############################################################################
+# We can observe that the "embarked" and "sex" columns were tagged as
+# "categorical" columns when calling ``fetch_openml``. Therefore, we can use
+# this information to dispatch the categorical columns to the
+# ``categorical_transformer`` and the remaining columns to the
+# ``numerical_transformer``.
+# Note that in practise, it is hope to you to affect the "categorical" dtype
+# to some specific columns.
+
+from sklearn.compose import make_column_selector as selector
+from pandas.api.types import CategoricalDtype
```

Comment:
We might be able to simplify to `from pandas import CategoricalDtype` depending on the version used. However, we are supporting a too old version for the moment.

Reply:
Or maybe it was `pd.Categorical` I don't recall exactly now.